### PR TITLE
Circular import cleanup

### PR DESCRIPTION
**Pull request by bpkroth**
Some circular import cleanup in preparation for config reorgs and importlib based loading from #313

### CODE REVIEW

File: mlos_bench/mlos_bench/environments/base_environment.py [hunk 1]
```diff
@@ -140,6 +138,11 @@ def __init__(self,
             _LOG.debug("Config for: %s\n%s",
                        name, json.dumps(self.config, indent=2))
 
+    @property
+    def _config_loader_service(self) -> "SupportsConfigLoading":
+        assert self._service is not None
+        return self._service.config_loader_service
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/environments/base_environment.py:144]: Slight tweak: avoids the need to store an extra reference.

File: mlos_bench/mlos_bench/environments/composite_env.py [hunk 2]
```diff
@@ -70,7 +70,7 @@ def _add_child(self, env: Environment) -> None:
         This method is called from the constructor only.
         """
         self._children.append(env)
-        self._tunable_params.update(env.tunable_params())
+        self._tunable_params.update(env.tunable_params)
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/environments/composite_env.py:73]: Cosmetic fixup

**Comment by bpkroth** [comment]
Seems to be a flaky test (`test_optimizer_with_llamatune`), but nothing I can see that changed that should affect that and is running fine in a loop on my local machine.

**Comment by motus** [comment]
> Seems to be a flaky test (`test_optimizer_with_llamatune`), but nothing I can see that changed that should affect that and is running fine in a loop on my local machine.

yes, I've seen it once or twice before - an optimizer recommends a value that falls out of the expected range. Is that the one? Anyway, I think I'm fine with restarting a CI job for now; we'll debug the issue later

**Review by motus — APPROVED** [review]
Looks good!

**Pull request merged**